src/wp-includes/block-editor.php

```diff
@@ -233,23 +226,9 @@ function get_default_block_editor_settings() {
 		'__unstableGalleryWithImageBlocks' => true,
 	);
 
-	// Theme settings.
-	$color_palette = current( (array) get_theme_support( 'editor-color-palette' ) );
-	if ( false !== $color_palette ) {
-		$editor_settings['colors'] = $color_palette;
-	}
+	$theme_settings = get_legacy_theme_supports_block_editor_settings();
 
-	$font_sizes = current( (array) get_theme_support( 'editor-font-sizes' ) );
-	if ( false !== $font_sizes ) {
-		$editor_settings['fontSizes'] = $font_sizes;
-	}
-
-	$gradient_presets = current( (array) get_theme_support( 'editor-gradient-presets' ) );
-	if ( false !== $gradient_presets ) {
-		$editor_settings['gradients'] = $gradient_presets;
-	}
-
-	return $editor_settings;
+	return array_merge( $editor_settings, $theme_settings );
```

Comment:
Let's avoid using `array_merge` due to its performance overhead. This came up in a few other performance optimization PRs before.
```suggestion
	foreach ( $theme_settings as $key => $value ) {
		$editor_settings[ $key ] = $value;
	}

	return $editor_settings;
```